Tests/DataPipeline/DataPipelineCompatibilityTests.swift

```diff
@@ -0,0 +1,438 @@
+    private var storage: Storage!
+
+    private let deviceInfoMock = DeviceInfoMock()
+    private let eventBusHandlerMock = EventBusHandlerMock()
+    private let globalDataStoreMock = GlobalDataStoreMock()
+
+    override func setUp() {
+        super.setUp()
+
+        diGraphShared.override(value: dateUtilStub, forType: DateUtil.self)
+        diGraphShared.override(value: deviceInfoMock, forType: DeviceInfo.self)
+        diGraphShared.override(value: eventBusHandlerMock, forType: EventBusHandler.self)
+        diGraphShared.override(value: globalDataStoreMock, forType: GlobalDataStore.self)
+
+        customerIO = createCustomerIOInstance()
+
+        // setting up analytics for testing
+        analytics = customerIO.analytics
+        guard let analytics = analytics else {
+            fatalError("Analytics instance is nil. The SDK has been set up incorrectly.")
+        }
+        // wait for analytics queue to start emitting events
+        analytics.waitUntilStarted()
+        // get storage instance so we can read final events
+        storage = analytics.storage
+        storage.hardReset(doYouKnowHowToUseThis: true)
+    }
+
+    override func tearDown() {
+        customerIO.clearIdentify()
+        super.tearDown()
+    }
+
+    // MARK: profile
+
+    func test_identifyWithoutAttributes_expectFinalJSONHasCorrectKeysAndValues() {
+        let givenIdentifier = String.random
+
+        customerIO.identify(identifier: givenIdentifier)
+
+        let allEvents = readTypeFromStorage(key: Storage.Constants.events)
+        let filteredEvents = allEvents.filter { $0.eventType == "identify" }
+        XCTAssertEqual(filteredEvents.count, 1, "too many events received")
+
+        guard let savedEvent = filteredEvents.last else {
+            XCTFail("saved event must not be nil")
+            return
+        }
+
+        XCTAssertEqual(savedEvent[keyPath: "userId"] as? String, givenIdentifier)
+        XCTAssertTrue(savedEvent[mapKeyPath: "traits"]?.isEmpty ?? false)
+    }
+
+    func test_identifyWithAttributes_expectFinalJSONHasCorrectKeysAndValues() {
+        let givenIdentifier = String.random
+        let givenBody: [String: Any] = ["first_name": "Dana", "age": 30]
+
+        customerIO.identify(identifier: givenIdentifier, body: givenBody)
+
+        let allEvents = readTypeFromStorage(key: Storage.Constants.events)
+        let filteredEvents = allEvents.filter { $0.eventType == "identify" }
+        XCTAssertEqual(filteredEvents.count, 1, "too many events received")
+
+        guard let savedEvent = filteredEvents.last else {
+            XCTFail("saved event must not be nil")
+            return
+        }
+
+        XCTAssertEqual(savedEvent[keyPath: "userId"] as? String, givenIdentifier)
+        XCTAssertMatches(
+            savedEvent[mapKeyPath: "traits"],
+            givenBody,
+            withTypeMap: [["age"]: Int.self]
+        )
+    }
+
+    func test_eventBeforeIdentify_expectFinalJSONHasNoUserId() {
```

Comment:
should we verify if there is an anonymous id?

Reply:
Updated